autoload/SpaceVim/api/prompt.vim

```diff
@@ -76,13 +76,24 @@ func! s:self._handle_input(...) abort
     endif
     call self._build_prompt()
   endif
+  let self.c_r_mode = 0
   while self._quit == 0
     let char = self._getchar()
     if has_key(self._function_key, char)
       call call(self._function_key[char], [])
       continue
     endif
-    if char ==# "\<Right>"
+    if self.c_r_mode ==# 1
+      let reg = '@' . char
+      let paste = eval(reg)
```

Comment:
need to check the char. it only can be register.

btw, we can use timer to set c_r_mode to 0